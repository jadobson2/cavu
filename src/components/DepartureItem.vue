<template>
  <article
    class="
      departure-item grid items-center gap-1 md:gap-2 xl:gap-3 mb-5 last:mb-0 px-8
      lg:px-14 py-5 border-2 rounded-xl text-xl xl:text-2xl text-white font-bold
      whitespace-nowrap
    "
    data-test="departure"
  >
    <span data-label="Departure time:">{{ departureTime }}</span>
    <span class="text-yellow" data-label="City name:">
      {{ departure.arrivalAirport.cityName }}
    </span>
    <span data-label="Code:">{{ departure.arrivalAirport.code }}</span>
    <span data-label="Airline:">{{ departure.airline.name }}</span>
    <span class="text-yellow" data-label="Gate:" data-test="departure-gate">
      {{ departure.departureGate ? departure.departureGate.number : '' }}
    </span>
    <DepartureStatus
      :status-text="departure.status"
      class="col-span-full lg:col-auto mt-3 md:mt-2 lg:mt-0"
    />
  </article>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'
import { Departure } from '@/types/index'
import DepartureStatus from './DepartureStatus.vue'
import { getHours } from '@/utils/date'

export default Vue.extend({
  name: 'DepartureItem',
  components: { DepartureStatus },
  props: {
    departure: {
      type: Object as PropType<Departure>
    }
  },
  computed: {
    departureTime (): string {
      return getHours(this.departure.scheduledDepartureDateTime)
    }
  }
})
</script>

<style lang="scss" scoped>
.departure-item {
  @media (min-width: #{theme('screens.md')}) {
    grid-template-columns: 15% 25% 15% 25% 15% 0%;
  }

  @media (min-width: #{theme('screens.lg')}) {
    grid-template-columns: 15% 20% 10% 20% 10% 30%;
  }

  @media (max-width: #{theme('screens.md')}) {
    [data-label] {
      @apply before:mr-1 before:content-[attr(data-label)];
    }
  }
}
</style>
