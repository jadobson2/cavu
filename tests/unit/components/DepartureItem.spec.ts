import { shallowMount } from '@vue/test-utils'
import DepartureItem from '@/components/DepartureItem.vue'

describe('DepartureItem', () => {
  const departure = {
    flightDirection: 'departure',
    scheduledDepartureDateTime: '2022-08-22T10:30:00',
    scheduledArrivalDateTime: '2022-08-22T01:00:00',
    estimatedDepartureDateTime: '2022-08-22T10:51:00',
    actualDepartureDateTime: '2022-08-22T10:49:00',
    arrivalAirport: {
      name: 'Charles de Gaulle Airport',
      cityName: 'Paris',
      countryName: 'France',
      code: 'CDG'
    },
    departureAirport: {
      name: 'Manchester Airport',
      cityName: 'Manchester',
      countryName: 'United Kingdom',
      code: 'MAN'
    },
    flightNumber: 'AF1669',
    airline: {
      name: 'Air France',
      code: 'AF'
    },
    departureGate: {
      name: 'Gate A5',
      number: 'A5',
      action: 'Final Call'
    },
    arrivalTerminal: null,
    departureTerminal: '2',
    status: 'Departed 10:49'
  }

  it('renders correctly', () => {
    const wrapper = shallowMount(DepartureItem, {
      propsData: { departure }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  test('no departure gate', () => {
    const wrapper = shallowMount(DepartureItem, {
      propsData: {
        departure: {
          ...departure,
          departureGate: null
        }
      }
    })

    expect(wrapper.find('[data-test="departure-gate"]').text()).toBe('')
  })
})
