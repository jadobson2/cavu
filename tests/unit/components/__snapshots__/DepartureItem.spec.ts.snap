// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepartureItem renders correctly 1`] = `
<article
  class="departure-item grid items-center gap-1 md:gap-2 xl:gap-3 mb-5 last:mb-0 px-8 lg:px-14 py-5 border-2 rounded-xl text-xl xl:text-2xl text-white font-bold whitespace-nowrap"
  data-test="departure"
>
  <span
    data-label="Departure time:"
  >
    10:30
  </span>
   
  <span
    class="text-yellow"
    data-label="City name:"
  >
    
    Paris
    
    <span
      class="block mt-1 md:mt-0 md:text-sm text-white"
      data-label="Country:"
    >
      
      France
    
    </span>
  </span>
   
  <span
    data-label="Code:"
  >
    CDG
  </span>
   
  <span
    data-label="Airline:"
  >
    Air France
  </span>
   
  <span
    class="text-yellow"
    data-label="Gate:"
    data-test="departure-gate"
  >
    
    A5
  
  </span>
   
  <departurestatus-stub
    class="col-span-full lg:col-auto mt-3 md:mt-2 lg:mt-0"
    statustext="Departed 10:49"
  />
</article>
`;
